VirtualEconomyFramework/VENFTApp-Blazor/wwwroot/jsInterop.js

```diff
@@ -292,91 +312,117 @@ window.jsFunctions = {
+
+                        var imageData = ctx.getImageData(0, 0, canvas.width, canvas.height);
                         var code = jsQR(imageData.data, imageData.width, imageData.height, {
                             inversionAttempts: "dontInvert",
-                        });
-
-                        if (code) {
-                            console.log("Found QR code", code.data);
-
-                            if (dotnetRef) {
-                                dotnetRef.invokeMethodAsync('ReturnBarcodeResultsAsync', code.data);
-                            }
-                        }
-                        else {
-                            if (dotnetRef) {
-                                dotnetRef.invokeMethodAsync('ReturnBarcodeResultsAsync', 'Cannot Load QR Code!');
+                        });
+
+                        if (code) {
+                            console.log("Found QR code", code.data);
+
+                            if (dotnetRef) {
+                                dotnetRef.invokeMethodAsync('ReturnBarcodeResultsAsync', code.data);
+                            }
+                        }
+                        else {
+                            if (dotnetRef) {
+                                dotnetRef.invokeMethodAsync('ReturnBarcodeResultsAsync', 'Cannot Load QR Code!');
                             }
                             console.log("No code found");
-                        }
-                    };
-                }*/
-            } catch (error) {
-                if (dotnetRef) {
-                    dotnetRef.invokeMethodAsync('ReturnBarcodeResultsAsync', `Cannot Load QR Code! ${error}`);
-                }
-            }
-
-        };
-        input.click();
-    },
-};
+                        }
+                    };
+                }*/
+            } catch (error) {
+                if (dotnetRef) {
+                    dotnetRef.invokeMethodAsync('ReturnBarcodeResultsAsync', `Cannot Load QR Code! ${error}`);
+                }
+            }
+
+        };
+        input.click();
+    },
+    modal: {
+        add: function (element) {
+            const container = document.getElementById("modal-container");
+            if (element && container) {
+                container.appendChild(element);
+            }
+        },
+        dispose: async function (element) {
+            if (!element || !element instanceof Element)
+                return;
+            handleModalClose(element);
+            element.remove();
+        },
+        open: function (element) {
+            if (!element || !element instanceof Element)
+                return;
+            element.classList.add("open");
+            handleModalOpen(element);
+        },
+        close: function (element) {
+            if (!element || !element instanceof Element)
+                return;
+            element.classList.remove("open");
+            handleModalClose(element);
+        },

```

Comment:
Triggers open/close animation and and handles scrollbar. Maybe we could also set `display: none;` when modal is not visible